src/cmd/package.go

```diff
@@ -336,20 +342,76 @@ func (o *packageMirrorResourcesOptions) run(cmd *cobra.Command, args []string) (
 		err = errors.Join(err, pkgLayout.Cleanup())
 	}()
 
-	mirrorOpt := packager2.MirrorOptions{
-		Cluster:               c,
-		PkgLayout:             pkgLayout,
-		RegistryInfo:          pkgConfig.InitOpts.RegistryInfo,
-		GitInfo:               pkgConfig.InitOpts.GitServer,
-		NoImageChecksum:       pkgConfig.MirrorOpts.NoImgChecksum,
-		Retries:               pkgConfig.PkgOpts.Retries,
-		OCIConcurrency:        config.CommonOptions.OCIConcurrency,
-		PlainHTTP:             config.CommonOptions.PlainHTTP,
-		InsecureSkipTLSVerify: config.CommonOptions.InsecureSkipTLSVerify,
-	}
-	err = packager2.Mirror(ctx, mirrorOpt)
-	if err != nil {
-		return err
+	images, repos := 0, 0
+	// Let's count the images and repos in the package
+	for _, component := range pkgLayout.Pkg.Components {
+		images += len(component.Images)
+		repos += len(component.Repos)
+	}
+	logger.From(ctx).Debug("Package contains images and repos", "images", images, "repos", repos)
+
+	// We don't yet know if the targets are internal or external
+	c, _ := cluster.New(ctx) //nolint:errcheck
+
+	if o.imagesOnly {
```

Comment:
nit: a common case will be that both `ImageOnly` and `reposOnly` are true, which I feel could make these names misleading. Thoughts on re-naming to something like `o.mirrorImages` and `o.mirrorRepos`?